diff --git a/ckeditor_link.module b/ckeditor_link.module
index 5bbd191..9305c8b 100644
--- a/ckeditor_link.module
+++ b/ckeditor_link.module
@@ -156,7 +156,7 @@ function ckeditor_link_ckeditor_link_types() {
 function ckeditor_link_element_info_alter(&$type) {
   if (user_access('access ckeditor link')) {
     $type['text_format']['#pre_render'][] = 'ckeditor_link_text_format_pre_render';
-    $type['form']['#post_render'][] = 'ckeditor_link_form_post_render';
+    $type['text_format']['#post_render'][] = 'ckeditor_link_text_format_post_render';
   }
 }
 
@@ -173,7 +173,7 @@ function _ckeditor_link_has_text_format($set = FALSE) {
   $has = TRUE;
 }
 
-function ckeditor_link_form_post_render($content, $element) {
+function ckeditor_link_text_format_post_render($content, $element) {
   static $added;
   if (!isset($added) && _ckeditor_link_has_text_format() && ($js = drupal_add_js()) && isset($js['settings']['data'])) {
     $settings = call_user_func_array('array_merge_recursive', $js['settings']['data']);
